### PR TITLE
Add iostat module

### PR DESCRIPTION
**Pull request by Ostrokrzew**
Needed to check IO statistics regardless of cache statistics, e.g. before cache start or after cache stop.

Signed-off-by: Slawomir Jankowski <slawomir.jankowski@intel.com>

### CODE REVIEW

File: test_tools/iostat.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+class IostatStatsEnum(IntEnum):
+    tps = 2,
+    blk_rdps = 3,
+    blk_wrtps = 4,
+    blk_reads = 5,
+    blk_writes = 6
+
+
+def get_iostat(device: Device, stat_filter: IostatStatsEnum = None):
+    """Returns one of basic statistics displayed in kB/ kBps as a float number."""
```
Comment (Deixx) [inline test_tools/iostat.py]: remove space after /

File: test_tools/iostat.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    output = TestRun.executor.run(f"iostat -dk {device.system_path}"
+                                  f" | grep {device.system_path.replace('/dev/', '')}"
+                                  f" | awk 'NR==1 {{print ${stat_filter.value}}}'").stdout
+    return float(output)
+
+
+def get_iostat_all(device: Device):
+    """Returns list of basic statistics displayed in kB/ kBps."""
```
Comment (Deixx) [inline test_tools/iostat.py]: the space

File: test_tools/iostat.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+from core.test_run import TestRun
+from storage_devices.device import Device
+
+
+class IostatStatsEnum(IntEnum):
+    tps = 2,
+    blk_rdps = 3,
+    blk_wrtps = 4,
+    blk_reads = 5,
+    blk_writes = 6
```
Comment (Deixx) [inline test_tools/iostat.py]: why blk if those are really kB?

File: test_tools/iostat.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+from aenum import IntEnum
+
+from core.test_run import TestRun
+from storage_devices.device import Device
+
+
+class IostatStatsEnum(IntEnum):
```
Comment (Deixx) [inline test_tools/iostat.py]: we know it's an Enum - what does it enumerate? BUT FIRST! - read my comment to line 25

File: test_tools/iostat.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    blk_reads = 5,
+    blk_writes = 6
+
+
+def get_iostat(device: Device, stat_filter: IostatStatsEnum = None):
+    """Returns one of basic statistics displayed in kB/ kBps as a float number."""
+    output = TestRun.executor.run(f"iostat -dk {device.system_path}"
+                                  f" | grep {device.system_path.replace('/dev/', '')}"
+                                  f" | awk 'NR==1 {{print ${stat_filter.value}}}'").stdout
+    return float(output)
```
Comment (Deixx) [inline test_tools/iostat.py]: I think methods here should return IoStats class which would have the output parsed to float properties. So you can call for example:
`get_iostat(device).tps`
this way we probably would need just one method

File: test_tools/iostat.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+
+def get_iostat_all(device: Device):
```
Comment (Deixx) [inline test_tools/iostat.py]: Why would you return float from one method and string[] from the other?

File: test_tools/iostat.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+from core.test_run import TestRun
+from storage_devices.device import Device
+
+
+class IOstat:
+    def __init__(self, stat_list: list = None):
+        if stat_list is None or len(stat_list) < 5:
+            stat_list.extend([0, 0, 0, 0, 0])
+        if len(stat_list) > 5:
+            del stat_list[5:]
```
Comment (robertbaldyga) [inline test_tools/iostat.py]: But actually why? If we are interested only in first 5 elements, then let's just ignore rest of them. I see no reason to delete elements from the list.

Comment (Ostrokrzew) [reply test_tools/iostat.py]: okey

File: test_tools/iostat.py [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+#
+
+from core.test_run import TestRun
+from storage_devices.device import Device
+
+
+class IOstat:
+    def __init__(self, stat_list: list = None):
+        if stat_list is None or len(stat_list) < 5:
+            stat_list.extend([0, 0, 0, 0, 0])
```
Comment (robertbaldyga) [inline test_tools/iostat.py]: `AttributeError: 'NoneType' object has no attribute 'extend'`
If you really need to handle `None` here, you need to do it another way.

Comment (Ostrokrzew) [reply test_tools/iostat.py]: thanks for warning

File: test_tools/iostat.py [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+    def __init__(self, stat_list: list = None):
+        if stat_list is None:
+            stat_list = [0, 0, 0, 0, 0]
+        if len(stat_list) < 5:
+            stat_list.extend([0, 0, 0, 0, 0])
+        self.tps = float(stat_list[0])
+        self.read_kBps = float(stat_list[1])
+        self.write_kBps = float(stat_list[2])
+        self.kB_reads = float(stat_list[3])
+        self.kB_writes = float(stat_list[4])
```
Comment (mmichal10) [inline test_tools/iostat.py]: Instead of raw floats, you can parse it to Size()

**Pull request merged**